src/adaptation/massmatrix.jl

```diff
@@ -102,20 +102,50 @@ function WelfordVar(sz::Union{Tuple{Int},Tuple{Int,Int}}; kwargs...)
     return WelfordVar{Float64}(sz; kwargs...)
 end
 
-function Base.resize!(wv::WelfordVar, θ::AbstractVecOrMat{T}) where {T<:AbstractFloat}
-    if size(θ) != size(wv.var)
+function Base.resize!(wv::WelfordVar, θ::AbstractMatrix{T}) where {T<:AbstractFloat}
```

Comment:
IMO this function must be renamed, it's a misuse of `Base.resize!` as the second argument of `resize!` is supposed to be an integer. It should just be an internal function.

Reply:
Yeah, `resize!` here is not appropriate for the context

Reply:
@ErikQQY If one passes `length(θ)` (i.e. `resize!(wv::WelfordVar, length(θ)`), then we could keep the function name `Base.resize!`. I slightly prefer `Base.resize!` since it is more readable than `resize_with`, whose meaning is unclear from the function name alone.

Reply:
Commit https://github.com/TuringLang/AdvancedHMC.jl/pull/427/commits/b4e60f872eec1a09950ec42888d02297baec1004 extends `Base.resize!` to allow passing `resize!(wv::WelfordVar, size(θ))` so that we can have a unifying resizing and also reduce the allocations. Let me know if this way introduces any misinterpretation.